_posts/2023-06-28-Remilia-interview.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "Interview wth contributor Remilia Scarlet"
+author: beta-zilliani,Remilia
+summary: "<b>Music playback projects in Crystal, games engines, and a bit of Lisp"
+---
+<https://remilia.sdf.org/>
+
+<https://www.youtube.com/watch?v=UX2in-whUik>
+
+_Conversation between Beta Ziliani and Remilia Scarlet_
+## A little about you. Who's Remilia?
+
+Well, I'm an open-source developer who has worked with quite a few languages over the years. I mainly do command line tools, but recently I've been interested in audio programming as well.
+
+## You are a Lisper and a Crystalist. That's a strange pairing of languages! What got you into Lisp, and what got you into Crystal?
+
+My interest in Common Lisp started in the mid 2000s just out of curiosity. I kept reading about how great the language could be, and felt that the programs I was working on at the time could benefit from it. So, I sat down, decided "I'm not stopping until I can write a non-trivial program in Lisp", and just went from there. Eventually it became one of my favorite languages. With Crystal, a few years ago I was looking for a language that could provide me with the performance I needed, but that could produce smaller binaries, and that could be easier for others should they want to contribute to my projects (I was the only one I knew who worked with Lisp at the time). After some Wikipedia searching, I discovered Crystal and found that it fit my needs perfectly.
+These days, I'd say they're both tied for my favorite language.
+## Sometimes you first do a prototype in Lisp, and then a rewrite in Crystal, is that right?
+
+Yes, that's correct.
```

Comment:
Oh that's very shortspoken. Would've been nice to get some more out of this.

Reply:
I can expand on this if you'd like.

Reply:
@MistressRemilia That would be nice. If you want to do it, feel free to paste it here or add it as a suggestion. Then, after that, we'll 🚀 it.

Reply:
Ok, new text is below the line.

----

Yes, that's correct.  Part of this is just due to my history of using Common Lisp.  I've gotten used to the immediacy of it, especially when combined with an Emacs mode called Slime.  You don't just have a REPL, but a live coding environment with the full compiler and image-based development.  I can just put my cursor in a function, do `C-c`, and it'll recompile a function to native code, even while a program is running.  So it makes the adjust->compile->test pipeline much shorter.  Common Lisp is close enough to Crystal with respect to its capabilities and paradigms that moving between them also isn't too big of a deal for me.

However, my Lisp code stays as a prototype in most cases because my Lisp code is never as memory efficient as Crystal.  In most cases I see a roughly 50-75% decrease in memory usage going from my initial Lisp prototype to Crystal.  Shipping a source repo that someone can download and compile also isn't as clean as with Shards, and even if it was, I don't think most users wanting to compile their own software know the idiosyncrasies of the various Lisp compilers out there.  So really, Crystal ends up making a lot more sense to me past the prototype stage in most cases.

Also, a Hello World binary built with Common Lisp is somewhere around 42mb (and can't be run through `strip`), whereas a Hello World in Crystal is only like 300k.